modules/standard/IO.chpl

```diff
@@ -3830,28 +3830,28 @@ proc channel.readHelper(ref args ...?k, style:iostyleInternal):bool throws {
   :arg amount: The maximum amount of bytes to read.
   :returns: true if the bytes were read without error.
 */
-proc channel.readline(arg: [] uint(8), out numRead : int, start = arg.domain.low,
-                      amount = arg.domain.high - start + 1) : bool throws
```

Comment:
What would you think about simply making this routine simply not work (halt) for arrays of negative strides for the time being?  Or, we could take a Michael suggestion and simply have it not work for strided arrays at all (which could be a compiler error).

Reply:
I think we can throw an IllegalArgumentError rather than halting on negative strides.

I'm OK with a compiler error for strided arrays for the time being. Long-term I'd like to make sure that whatever we have here is consistent with the rest of the standard APIs regarding strided arrays (not that you're suggesting otherwise).

Reply:
The other thing I worry about a bit here is that @ShreyasKhandekar has been refactoring these routines as part of his deprecation of `readline()` in its current form...  If it weren't for that, I'd be more inclined to wrestle with this further.

Reply:
Tagging @mppf on this discussion as well, as he advocated for this earlier as well (though perhaps more in the "we should never support it" sense, where I think it's more plausible that we should...).